dist/tools/compile_test/tests/test.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+COMPILE_TEST_DIR="$(readlink -f "$(dirname "$0")/..")"
+
+# prints a character representing the result:
+#   - a green tick when $1 is 0
+#   - a red cross when $1 is 1
+function print_result {
+    local RED="\033[0;31m"
+    local GREEN="\033[0;32m"
+    local NO_COLOUR="\033[0m"
+
+    if (( "$1" == 0 )); then
+        echo -e "${GREEN}✓$NO_COLOUR"
+    else
+        echo -e "${RED}x$NO_COLOUR"
+    fi
+}
+
+
+# sets the global accumulative RESULT variable to keep record of errors during
+# tests.
+function set_result {
+    NEW_RESULT=$1
+
+    if (( NEW_RESULT != 0))
+    then
+        RESULT=$NEW_RESULT
+    fi
+}
+
+# runs a test passed in $1. It will print the corresponding name and result and
+# update the global state of the tests.
+function run_test {
+    echo -n "Running test: \"$*\" "
+    $1
+    res=$?
+
+    print_result $res
+    set_result $res
+}
+
+function call_with_defaults {
+    if  OUT=$(${COMPILE_TEST_DIR}/compile_like_murdock.py -d 2>&1)
```

Comment:
use `--dry-run` to enhance the readability for (almost) zero costs!

Reply:
Ah, ofc, this also works with `--apps` and `--boards`

Reply:
Done